libsolidity/ast/ASTJsonExporter.cpp

```diff
@@ -430,6 +430,7 @@ bool ASTJsonExporter::visit(EnumValue const& _node)
 	setJsonNode(_node, "EnumValue", {
 		std::make_pair("name", _node.name()),
 		std::make_pair("nameLocation", sourceLocationToString(_node.nameLocation())),
+		std::make_pair("documentation", _node.documentation() ? toJson(*_node.documentation()) : Json()),
```

Comment:
Could also use the overload `toJson(ASTNode*)` which already tests and returns the proper value.

Reply:
@cameel unfortunately, this doesn't quite work in the case when a natspec doc is provided [beyond the last value](https://github.com/ethereum/solidity/pull/15956/files#r2024478891), as the natspec comment is not associated with any `_node`, which then segfaults [here](https://github.com/ethereum/solidity/blob/develop/libsolidity/ast/ASTJsonExporter.cpp#L209).

Going over the exporter in more detail now, and we perform a `node.documentation()` check for every single export, i.e. we don't use `toJson(*_node.documentation())` by itself.

I'll therefore keep this as is.

Reply:
No problem keeping it as is, but this works for me:
```suggestion
		std::make_pair("documentation", toJson(&*_node.documentation()))
```
or

```suggestion
		std::make_pair("documentation", toJson(_node.documentation().get()))
```

Reply:
Aaah OK, guess I'm blind having not seen the second `toJson()` overload that was right below the one I was debugging.